Generated interface files do not import dependencies

TL;DR
=====
* If `[ExportTs*]` targets a `class` - imports are specified correctly
* If `[ExportTs*]` targets an `interface` - imports are omitted

Problem
=======

I have something like this

```
# ISomeInterface.cs
[ExportTsInterface]
public interface ISomeInterface {
    IList<IOtherInterface> Others { get; set; }
}

# IOtherInterface.cs
[ExportTsInterface]
public interface IOtherInterface {

}
```

Generated files:

```
# i-some-interface.ts
# there's no "import { IOtherInterface } from './i-other-interface.ts';" here
export interface ISomeInterface {
   others: IOtherInterface[];
}
```

Typescript compiler now gives me an error
```
ERROR in src/i-some-interface.ts(12,17): error TS2304: Cannot find name 'IOtherInterface'
```

If I use `[ExportTsInterface]` on a class instead, `*.ts` files now correctly have `import { IOtherInterface } from ...` in them.

Environment
============
Nuget:
`<PackageReference Include="TypeGen" Version="2.4.7" />`
Tool:
```
"dotnet-typegen": {
      "version": "2.4.7",
      "commands": [
        "dotnet-typegen"
      ]
    }
```
`tgconfig`:
```
{
  "assemblies": [
    "src/web/bin/Release/netcoreapp3.0/web.dll",
    "src/web/bin/Release/netcoreapp3.0/core.dll",
    "src/web/bin/Release/netcoreapp3.0/infrastructure.dll"
  ],
  "clearOutputDirectory": true,
  "outputPath": "src/front-end/src/models",
  "createIndexFile": true,
  "enumStringInitializersConverters": [],
  "customTypeMappings": {
    "NodaTime.Instant": "Date"
  }
}
```

--- a/src/TypeGen/TypeGen.Core/Generator/Generator.cs
+++ b/src/TypeGen/TypeGen.Core/Generator/Generator.cs
@@ -376,12 +376,18 @@ namespace TypeGen.Core.Generator
         /// <returns>Generated TypeScript file paths (relative to the Options.BaseOutputDirectory)</returns>
         private IEnumerable<string> GenerateNotMarked(Type type, string outputDirectory)
         {
-            if (type.GetTypeInfo().IsClass)
+            var typeInfo = type.GetTypeInfo();
+            if (typeInfo.IsClass)
             {
                 return GenerateClass(type, new ExportTsClassAttribute { OutputDir = outputDirectory });
             }
 
-            if (type.GetTypeInfo().IsEnum)
+            if (typeInfo.IsInterface)
+            {
+                return GenerateInterface(type, new ExportTsInterfaceAttribute { OutputDir = outputDirectory });
+            }
+
+            if (typeInfo.IsEnum)
             {
                 return GenerateEnum(type, new ExportTsEnumAttribute { OutputDir = outputDirectory });
             }

--- a/src/TypeGen/TypeGen.Core/Generator/Services/TypeDependencyService.cs
+++ b/src/TypeGen/TypeGen.Core/Generator/Services/TypeDependencyService.cs
@@ -35,8 +35,10 @@ namespace TypeGen.Core.Generator.Services
         public IEnumerable<TypeDependencyInfo> GetTypeDependencies(Type type)
         {
             Requires.NotNull(type, nameof(type));
+
+            var typeInfo = type.GetTypeInfo();
             
-            if (!type.GetTypeInfo().IsClass) return Enumerable.Empty<TypeDependencyInfo>();
+            if (!typeInfo.IsClass && !typeInfo.IsInterface) return Enumerable.Empty<TypeDependencyInfo>();
 
             type = _typeService.StripNullable(type);
 

